public/app/features/trails/MetricSelectScene.tsx

```diff
@@ -273,18 +279,31 @@ export class MetricSelectScene extends SceneObjectBase<MetricSelectSceneState> {
     this.buildLayout();
   };
 
+  public onDataSourceChange() {
+    // Clear filtered metrics
+    this.setState({ metricsAfterFilter: undefined, metricsAfterSearch: undefined });
+    // Clear all panels for the previous data source
+    this.state.body.setState({ children: [] });
+
+    // Cancel any ongoing queries for metric names.
+    const variable = sceneGraph.lookupVariable(VAR_METRIC_NAMES, this);
+    if (variable instanceof QueryVariable) {
+      variable.onCancel();
+    }
```

Comment:
hm.. interesting, this should technically be handled automatically I think. 
Created an issue in scenes: https://github.com/grafana/scenes/issues/556